ffi/src/test_ffi.rs

```diff
@@ -0,0 +1,100 @@
+//! Utility functions used for testing ffi code
+
+use std::{ops::Not, sync::Arc};
+
+use crate::{expressions::SharedExpression, handle::Handle};
+use delta_kernel::{
+    expressions::{ArrayData, BinaryOperator, Expression, Scalar, StructData},
+    schema::{ArrayType, DataType, PrimitiveType, StructField, StructType},
+};
+
+/// Constructs a kernel expression that is passed back as a SharedExpression handle
+///
+/// # Safety
+/// The caller is responsible for freeing the retured memory, either by calling
+/// [`free_kernel_predicate`], or [`Handle::drop_handle`]
+#[no_mangle]
+pub unsafe extern "C" fn get_testing_kernel_expression() -> Handle<SharedExpression> {
+    use Expression as Expr;
+
+    let array_type = ArrayType::new(
+        DataType::Primitive(delta_kernel::schema::PrimitiveType::Short),
+        false,
+    );
+    let array_data = ArrayData::new(array_type.clone(), vec![Scalar::Short(5), Scalar::Short(0)]);
+
+    let nested_fields = vec![
+        StructField::new("a", DataType::Primitive(PrimitiveType::Integer), false),
+        StructField::new("b", DataType::Array(Box::new(array_type)), false),
```

Comment:
Not seeing any `DataType::array` method. I went I ahead and implemented it.

Reply:
Oh sorry... it's `DataType::from(array_type)` or `array_type.into()`. But since [StructField::new](https://github.com/delta-incubator/delta-kernel-rs/blob/main/kernel/src/schema.rs#L98) accepts `impl Into<DataType>`, can just pass the `array_type` directly!